drivers/rtc/rtc_ds3231.c

```diff
@@ -0,0 +1,832 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Gergo Vari <work@varigergo.hu>
+ */
+
+/* TODO: make sure we're in 24h mode */
+/* TODO: implement configurable settings */
+/* TODO: implement get_temp */
```

Comment:
You need to look at devices that are implementing MFD (multifunction device). Essentially you would have a different *sensor* driver for what concerns the temperature, and the mfd compatible would allow you to lump the RTC and sensor under the same DT node.
See ex. https://github.com/zephyrproject-rtos/zephyr/blob/143b14bb4fb779437fdeb8369e42fbb81db4a5fd/boards/m5stack/m5stack_core2/m5stack_core2_procpu.dts#L119 for an MFD that acts as both a regulator and GPIO controller

Reply:
Was about to ask this question now! Thank you for pointing me in the right direction.